src/compiler/checker.ts

```diff
@@ -6420,14 +6440,14 @@ namespace ts {
             }
         }
 
-        function containsArgumentsReference(declaration: FunctionLikeDeclaration): boolean {
+        function containsArgumentsReference(declaration: SignatureDeclaration): boolean {

```

Comment:
`containsArgumentReference` is only called once, and the caller did the cast before. Since the same assumption gets made in the same order either way, I moved it to shorten an overlong line.